Lib/test/test_builtin.py

```diff
@@ -1511,6 +1513,126 @@ def test_construct_singletons(self):
             self.assertRaises(TypeError, tp, 1, 2)
             self.assertRaises(TypeError, tp, a=1, b=2)
 
+
+class TestBreakpoint(unittest.TestCase):
+    def test_breakpoint(self):
+        with patch('pdb.set_trace') as mock:
+            breakpoint()
+        mock.assert_called_once()
+
+    def test_breakpoint_with_breakpointhook_set(self):
+        call_status = 'Not called'
+        def my_breakpointhook():
+            nonlocal call_status
+            call_status = 'Called'
+        try:
+            sys.breakpointhook = my_breakpointhook
+            breakpoint()
+        finally:
+            sys.breakpointhook = sys.__breakpointhook__
+        self.assertEqual(call_status, 'Called')
+
+    def test_breakpoint_with_breakpointhook_reset(self):
+        call_status = 'Not called'
+        def my_breakpointhook():
+            nonlocal call_status
+            call_status = 'Called'
+        try:
+            sys.breakpointhook = my_breakpointhook
+            breakpoint()
+        finally:
+            sys.breakpointhook = sys.__breakpointhook__
+        self.assertEqual(call_status, 'Called')
+        with patch('pdb.set_trace') as mock:
+            breakpoint()
+        mock.assert_called_once()
+
+    def test_breakpoint_with_args_and_keywords(self):
+        call_args = None
+        def my_breakpointhook(*args, **kws):
+            nonlocal call_args
+            call_args = args, kws
+        try:
+            sys.breakpointhook = my_breakpointhook
+            breakpoint(1, 2, 3, four=4, five=5)
+        finally:
+            sys.breakpointhook = sys.__breakpointhook__
+        self.assertEqual(call_args, ((1, 2, 3), dict(four=4, five=5)))
+
+    def test_breakpoint_with_passthru_error(self):
+        def my_breakpointhook():
+            pass
+        try:
+            sys.breakpointhook = my_breakpointhook
+            self.assertRaises(TypeError, breakpoint, 1, 2, 3, four=4, five=5)
+        finally:
+            sys.breakpointhook = sys.__breakpointhook__
+
+    def test_envar_good_path_builtin(self):
+        with ExitStack() as resources:
+            env = resources.enter_context(EnvironmentVarGuard())
+            env['PYTHONBREAKPOINT'] = 'int'
+            mock = resources.enter_context(patch('builtins.int'))
+            breakpoint('7')
+            mock.assert_called_once_with('7')
+
+    def test_envar_good_path_other(self):
```

Comment:
Would it be possible to add a second V, I'm confused by "envar" for "env var" :-)

Reply:
Oh, ha!  "envvar" always looks weird to me.  I've always read/written the shortcut for "environment variable" as "envar" so that's just a habit I always have to break when using `:envvar:` for example. :)

Both are reasonable shortcuts, e.g. **env**ironment **var**iable vs. **en**vironment **var**iable.

If you don't mind, I'll keep it this way.

Reply:
I don't mind.

Reply:
What madman shortens environment to en? :)

Reply:
Ha ha, I proposed the same change, but @warsaw likes "envar" with a single V. It looks a typo to me, but @warsaw likes it. Honestly, since it's just an unit test, I don't care ;-)